### DISCUSSION TITLE
Citation keys can't combine first name and last name of Chinese.

### DISCUSSION THREAD

**Discussion by earthmoon** [retorquere/zotero-better-bibtex#2380] [Q&A] [answered]
Hello,
Using authors' names and years to make citation keys, Better Bibtex can't combine the first and last names if the Chinese name is separated into the first and last names. Only the last name in citation keys has no meaning as the last name is duplicated too many and the last name is only one character, which is too short.
May you make it combine last name and first name based on the Chinese?
Thanks.

**Comment by retorquere** [comment]
I'll need a debug log for that.

**Reply by retorquere** [reply]
But this is a change request, please open an issue for change requests

**Comment by earthmoon** [comment, marked as answer]
> x can't combine the first and last names if the Chinese name is separated into the first and last names. Only the last name in citation keys has no meaning as the last name is duplicated too many and the last name is only one character, which is too short.

Thanks. The debug ID is: ERHRLB2T-refs-apse. 
![image](https://user-images.githubusercontent.com/45897471/212538640-e740558a-3870-473e-98ac-aee08211300e.png)
When the first name and last name are separated, Better Bibtex always uses the last name. However, the 1.4 billion Chinese only have nearly 100 last names, so there is no significant difference between last names. When there are Chinese, using last name + first name is better.

**Reply by retorquere** [reply]
There's to my knowledge not a reliable way to detect Chinese as the ideaographs overlap with Japanese and Korean, but if you set the `language` field of the item to `zh`, `chinese` or `中文`, this should work:

```
language(chinese) + authors(1, name='%(g)s%(f)s') + year | auth.lower + year
```

**Reply by earthmoon** [reply]
Thanks for your kind help. Every time I ask some questions about setting citation keys, you can give me some codes to solve them. What should I learn to make these codes myself to meet my requirements? And the codes: `language(chinese) + authors(1, name='%(g)s%(f)s') + year | auth.lower + year` will set the first name before the last name. How to make the last name before the first name?
Thanks.

**Reply by retorquere** [reply]
> What should I learn to make these codes myself to meet my requirements?

Documentation for the citekey formulas is at https://retorque.re/zotero-better-bibtex/citing/, but it takes some time to get familiar with how these work. It's always OK to ask.

> How to make the last name before the first name?

`authors(1, name='%(f)s%(g)s')`

**Reply by earthmoon** [reply]
> > What should I learn to make these codes myself to meet my requirements?
> 
> Documentation for the citekey formulas is at https://retorque.re/zotero-better-bibtex/citing/, but it takes some time to get familiar with how these work. It's always OK to ask.
> 
> > How to make the last name before the first name?
> 
> `authors(1, name='%(f)s%(g)s')`

Thanks. I should learn JavaScript in the future.

**Reply by retorquere** [reply]
No worries, and this doesn't require javascript knowledge, but the behavior of a [specific library](https://github.com/alexei/sprintf.js). But it is always OK to ask.

**Reply by earthmoon** [reply]
> No worries, and this doesn't require javascript knowledge, but the behavior of a [specific library](https://github.com/alexei/sprintf.js). But it is always OK to ask.

Thanks a lot. I will read the content of that link.

**Reply by retorquere** [reply]
I honestly wouldn't spend too much time on the library. What's important to know is that `%(f)s` gives you the family name etc. The library allows for fancy stuff most people have no use for.

**Reply by earthmoon** [reply]
> ancy stuff

Yes. Thanks for your advice. Solving the problems with the knowledge needed is enough. No need to learn too much things that will be forgotten in a few days.
